File: try/prime.scala
import scala.math.sqrt
import java.time.Instant

object Main {
  def isPrime(n: Int): Boolean = {
    if (n <= 1) return false

    val end = sqrt(n).toInt
    for (i <- 2 to end) {
      if (n % i == 0) return false
    }
    true
  }

  def main(args: Array[String]): Unit = {
    val start = Instant.now.toEpochMilli
    var c = 0

    for (i <- 0 until 9000000) {
      if (isPrime(i)) c += 1
    }
    println(c)
    val end = Instant.now.toEpochMilli
    println(s"${end - start}ms")
  }
}

File: try/prime.zig
const std = @import("std");

fn is_prime(n: u32) bool {
    if (n <= 1) return false;
    var i: u32 = 2;
    const end = @as(u32, @intFromFloat(@sqrt(@as(f32, @floatFromInt(n)))));
    while (i <= end) : (i += 1) {
        if (n % i == 0) return false;
    }
    return true;
}

pub fn main() !void {
    const start = std.time.milliTimestamp();
    var c: u32 = 0;
    var i: u32 = 0;
    while (i < 9000000) : (i += 1) {
        if (is_prime(i)) {
            c += 1;
        }
    }
    const end = std.time.milliTimestamp();
    const stdout = std.io.getStdOut().writer();
    try stdout.print("{d}\n", .{c});
    try stdout.print("{d}ms\n", .{end - start});
}
